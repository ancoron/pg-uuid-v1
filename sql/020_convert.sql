SET timezone TO 'Zulu';
\x

-- conversion
SELECT pg_typeof(uuid_v1_convert('edb4d8f0-1a80-11e8-98d9-e03f49f7f8f3'::uuid)) AS from_std;
SELECT pg_typeof(uuid_v1_convert('edb4d8f0-1a80-11e8-98d9-e03f49f7f8f3'::uuid_v1)) AS to_std;
WITH data AS (
    SELECT
        uuid_v1_convert('607ad07c-f95a-11eb-adf0-9d8ba2d04971'::uuid) AS uuid_to_uuid_v1,
        '607ad07c-f95a-11eb-adf0-9d8ba2d04971'::uuid_v1 AS string_to_uuid_v1
)
SELECT
    uuid_to_uuid_v1,
    uuid_v1_get_timestamp(uuid_to_uuid_v1) AS conv_timestamp,
    uuid_v1_get_clockseq(uuid_to_uuid_v1) AS conv_clock_seq,
    string_to_uuid_v1,
    uuid_v1_get_timestamp(string_to_uuid_v1) AS string_timestamp,
    uuid_v1_get_clockseq(string_to_uuid_v1) AS string_clock_seq,
    (uuid_to_uuid_v1 = string_to_uuid_v1)::text AS equals_conversion_to_uuid_v1
FROM data
;
WITH data AS (
    SELECT
        uuid_v1_convert('7ca71896-f95a-11eb-adf0-9d8ba2d04971'::uuid_v1) AS uuid_v1_to_uuid,
        '7ca71896-f95a-11eb-adf0-9d8ba2d04971'::uuid AS string_to_uuid
)
SELECT
    uuid_v1_to_uuid,
    string_to_uuid,
    (uuid_v1_to_uuid = string_to_uuid)::text AS equals_conversion_to_uuid
FROM data
;
SELECT
    uuid_v1_convert(uuid_v1_convert('edb4d8f0-1a80-11e8-98d9-e03f49f7f8f3'::uuid_v1))
    AS double_convert
;
