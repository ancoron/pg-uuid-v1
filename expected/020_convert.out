SET timezone TO 'Zulu';
\x
-- conversion
SELECT pg_typeof(uuid_v1_convert('edb4d8f0-1a80-11e8-98d9-e03f49f7f8f3'::uuid)) AS from_std;
-[ RECORD 1 ]-----
from_std | uuid_v1

SELECT pg_typeof(uuid_v1_convert('edb4d8f0-1a80-11e8-98d9-e03f49f7f8f3'::uuid_v1)) AS to_std;
-[ RECORD 1 ]
to_std | uuid

WITH data AS (
    SELECT
        uuid_v1_convert('607ad07c-f95a-11eb-adf0-9d8ba2d04971'::uuid) AS uuid_to_uuid_v1,
        '607ad07c-f95a-11eb-adf0-9d8ba2d04971'::uuid_v1 AS string_to_uuid_v1
)
SELECT
    uuid_to_uuid_v1,
    uuid_v1_get_timestamp(uuid_to_uuid_v1) AS conv_timestamp,
    uuid_v1_get_clockseq(uuid_to_uuid_v1) AS conv_clock_seq,
    string_to_uuid_v1,
    uuid_v1_get_timestamp(string_to_uuid_v1) AS string_timestamp,
    uuid_v1_get_clockseq(string_to_uuid_v1) AS string_clock_seq,
    (uuid_to_uuid_v1 = string_to_uuid_v1)::text AS equals_conversion_to_uuid_v1
FROM data
;
-[ RECORD 1 ]----------------+-------------------------------------
uuid_to_uuid_v1              | 607ad07c-f95a-11eb-adf0-9d8ba2d04971
conv_timestamp               | Mon Aug 09 21:40:12.596646 2021 UTC
conv_clock_seq               | 11760
string_to_uuid_v1            | 607ad07c-f95a-11eb-adf0-9d8ba2d04971
string_timestamp             | Mon Aug 09 21:40:12.596646 2021 UTC
string_clock_seq             | 11760
equals_conversion_to_uuid_v1 | true

WITH data AS (
    SELECT
        uuid_v1_convert('7ca71896-f95a-11eb-adf0-9d8ba2d04971'::uuid_v1) AS uuid_v1_to_uuid,
        '7ca71896-f95a-11eb-adf0-9d8ba2d04971'::uuid AS string_to_uuid
)
SELECT
    uuid_v1_to_uuid,
    string_to_uuid,
    (uuid_v1_to_uuid = string_to_uuid)::text AS equals_conversion_to_uuid
FROM data
;
-[ RECORD 1 ]-------------+-------------------------------------
uuid_v1_to_uuid           | 7ca71896-f95a-11eb-adf0-9d8ba2d04971
string_to_uuid            | 7ca71896-f95a-11eb-adf0-9d8ba2d04971
equals_conversion_to_uuid | true

SELECT
    uuid_v1_convert(uuid_v1_convert('edb4d8f0-1a80-11e8-98d9-e03f49f7f8f3'::uuid_v1))
    AS double_convert
;
-[ RECORD 1 ]--+-------------------------------------
double_convert | edb4d8f0-1a80-11e8-98d9-e03f49f7f8f3

